.github/workflows/test_workflow_scripts/check-deprecated-deps.sh

```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+
+# Run `go list -m -u all` to list all dependencies.
+output=$(go list -m -u all)
+
+found_deprecated=false
+
+while IFS= read -r line; do
+    if [[ "$line" == *"deprecated"* ]]; then
+        echo "Deprecated dependency found: $line"
+        found_deprecated=true
+    fi
+done <<< "$output"
+
+if [ "$found_deprecated" = true ]; then
+    echo "Exiting with failure due to deprecated dependencies."
+    exit 1
+fi
```

Comment:
not just find ensure that we are deleting that dependency as well and then again run this script

Reply:
Ideally all the pipelines should pass after changes. 
Add that and you are GTG!

Reply:
I don't know how deleting the dependency will help as it will be used in our original code as a direct or indirect dependency.